### ISSUE TITLE
Constrast rule bug - IBMA_Color_Contrast_WCAG2AA

### ISSUE THREAD

**Issue by marcjohlic** [IBMa/equal-access#826]
Opening on behalf of Tom Babinzski/Essential Accessibility:

My colleague discovered a contrast rule bug, which we tend to see at many places throughout a site. Here is the explanation.
 

- URL: https://www.adrianpsych.com/
- Success Criterion in violation: 1.4.3 Contrast (Minimum) Level AA
- Rule ID: **IBMA_Color_Contrast_WCAG2AA**
- HTML Element: 
`<div data-variant="button_variant_17" data-elementtype="BUTTON2" class="element button_preset_1 " title="" id="vbid-e0f5c8aa-b3d2-46ad-9681-ef40324052f4-dyn_elem_BUTTON_0s2">`
- XPath or CSS Selector: 
`/html[1]/body[1]/div[2]/div[1]/div[1]/section[9]/div[1]/div[1]/article[1]/div[2]/div[1]/a[2]/div[1]`

This particular element is a button and the client has added the :after selector to create a pseudo-element to add cosmetic content to the element, in this case they have added a transparent background color and I believe this is why this is coming up as a false positive.
 
```
div[data-variant="button_variant_17"]::after {
    content: "→";
    position: relative;
    margin-left: 10px;
    font-family: "Arial";
    color: var(--button-color-preset-color) !important;
    font-size: 20px !important;
    background-color: transparent !important;
}
 
```

**Comment by marcjohlic** [comment]
From thbrunet:
>I don’t think it’s related to the ::after. Not sure why it’s miscalculating though.

**Comment by shunguoy** [comment]
reproducible, possibly related to the css color calculation

**Comment by shunguoy** [comment]
@philljenkins  please check the help for the rule IBMA_Color_Contrast_WCAG2AA for any update.

**Comment by shunguoy** [comment]
use the original page to test: https://www.adrianpsych.com/

**Comment by marcjohlic** [comment]
@tombrunet can you please review

**Comment by eirckr** [comment]
This seems to be fixed, tested with client site and not triggering
